--- boost_1_58_0/tools/build/src/tools/gcc.jam~	2015-04-04 18:25:07.000000000 +0100
+++ boost_1_58_0/tools/build/src/tools/gcc.jam	2015-04-19 23:09:59.044121516 +0100
@@ -451,7 +451,7 @@ rule setup-address-model ( targets * : s
         else
         {
             local arch = [ feature.get-values architecture : $(properties) ] ;
-            if $(arch) != arm
+            if $(arch) != arm && $(arch) != ""
             {
                 if $(model) = 32
                 {
