--- colord-1.1.4/lib/colord/cd-edid.c~	2013-10-30 14:19:28.000000000 +0000
+++ colord-1.1.4/lib/colord/cd-edid.c	2013-11-24 00:58:10.000000000 +0000
@@ -109,6 +109,7 @@ cd_edid_convert_pnp_id_to_string (const
 {
 	gchar *modalias = NULL;
 	gchar *vendor = NULL;
+#if 0
 	struct udev_hwdb *hwdb = NULL;
 	struct udev_list_entry *e;
 	struct udev_list_entry *v;
@@ -141,6 +142,7 @@ out:
 		udev_hwdb_unref (hwdb);
 	if (udev != NULL)
 		udev_unref (udev);
+#endif
 	return vendor;
 }
 
