--- gimp-2.8.8/plug-ins/common/file-mng.c~	2012-03-12 19:18:10.000000000 +0000
+++ gimp-2.8.8/plug-ins/common/file-mng.c	2013-11-24 10:52:44.000000000 +0000
@@ -53,11 +53,6 @@
 #endif
 
 
-/* libpng and jpeglib are currently used in this plug-in. */
-
-#include <png.h>
-#include <jpeglib.h>
-
 
 /* Grrr. The grrr is because the following have to be defined
  * by the application as well for some reason, although they
@@ -86,6 +81,12 @@
 
 #include <libmng.h>
 
+
+/* libpng and jpeglib are currently used in this plug-in. */
+
+#include <png.h>
+#include <jpeglib.h>
+
 #include "libgimp/gimp.h"
 #include "libgimp/gimpui.h"
 
--- gimp-2.8.8/app/text/gimpfont.c~	2012-05-08 07:05:46.000000000 +0100
+++ gimp-2.8.8/app/text/gimpfont.c	2014-03-09 21:44:09.000000000 +0000
@@ -27,7 +27,8 @@
 
 #define PANGO_ENABLE_ENGINE  1   /* Argh */
 #include <pango/pango-ot.h>
-#include <freetype/tttables.h>
+#include <ft2build.h>
+#include FT_TRUETYPE_TABLES_H
 
 #include "text-types.h"
 
