--- glibc-2.17/libio/stdio.h~	2012-12-25 03:02:13.000000000 +0000
+++ glibc-2.17/libio/stdio.h	2013-03-23 20:01:02.000000000 +0000
@@ -622,7 +622,7 @@ __BEGIN_NAMESPACE_STD
 extern char *fgets (char *__restrict __s, int __n, FILE *__restrict __stream)
      __wur;
 
-#if !defined __USE_ISOC11 \
+#if 1 || !defined __USE_ISOC11 \
     || (defined __cplusplus && __cplusplus <= 201103L)
 /* Get a newline-terminated string from stdin, removing the newline.
    DO NOT USE THIS FUNCTION!!  There is no limit on how much it will read.
 
