--- gmime-2.4.9/gmime/gmime-encodings.c~	2009-04-24 03:04:47.000000000 +0100
+++ gmime-2.4.9/gmime/gmime-encodings.c	2009-10-08 13:00:16.000000000 +0100
@@ -830,7 +830,7 @@ g_mime_encoding_uudecode_step (const uns
 						*outptr++ = gmime_uu_rank[b0] << 2 | gmime_uu_rank[b1] >> 4;
 						uulen--;
 					}
-					if (uulen >= 2) {
+					if (uulen >= 1) {
 						*outptr++ = gmime_uu_rank[b1] << 4 | gmime_uu_rank[b2] >> 2;
 						uulen--;
 					}
