--- modules/smb-method.c~	2005-06-08 08:19:54.000000000 +0100
+++ modules/smb-method.c	2005-09-09 11:46:07.000000000 +0100
@@ -1486,7 +1486,7 @@ do_close (GnomeVFSMethod *method,
 
 		/* Important: perform_authentication leaves and re-enters the lock! */
 		while (perform_authentication (&actx) > 0) {
-			r = smb_context->close (smb_context, handle->file);
+			r = smb_context->close_fn (smb_context, handle->file);
 			actx.res = (r >= 0) ? GNOME_VFS_OK : gnome_vfs_result_from_errno ();
 		}
 
