--- gst-plugins-base-1.5.2/gst-libs/gst/video/video-scaler.c~	2015-06-22 09:15:57.000000000 +0100
+++ gst-plugins-base-1.5.2/gst-libs/gst/video/video-scaler.c	2015-07-05 22:08:19.313860744 +0100
@@ -1316,14 +1316,14 @@ gst_video_scaler_horizontal (GstVideoSca
     gpointer src, gpointer dest, guint dest_offset, guint width)
 {
   gint n_elems;
-  GstVideoScalerHFunc func;
+  GstVideoScalerHFunc func = NULL;
 
   g_return_if_fail (scale != NULL);
   g_return_if_fail (src != NULL);
   g_return_if_fail (dest != NULL);
   g_return_if_fail (dest_offset + width <= scale->resampler.out_size);
 
-  if (!get_functions (scale, NULL, format, &func, NULL, &n_elems, &width))
+  if (!get_functions (scale, NULL, format, &func, NULL, &n_elems, &width) || !func)
     goto no_func;
 
   if (scale->tmpwidth < width)
@@ -1357,14 +1357,14 @@ gst_video_scaler_vertical (GstVideoScale
     gpointer src_lines[], gpointer dest, guint dest_offset, guint width)
 {
   gint n_elems;
-  GstVideoScalerVFunc func;
+  GstVideoScalerVFunc func = NULL;
 
   g_return_if_fail (scale != NULL);
   g_return_if_fail (src_lines != NULL);
   g_return_if_fail (dest != NULL);
   g_return_if_fail (dest_offset < scale->resampler.out_size);
 
-  if (!get_functions (NULL, scale, format, NULL, &func, &n_elems, &width))
+  if (!get_functions (NULL, scale, format, NULL, &func, &n_elems, &width) || !func)
     goto no_func;
 
   if (scale->tmpwidth < width)
@@ -1410,14 +1410,14 @@ gst_video_scaler_2d (GstVideoScaler * hs
     guint width, guint height)
 {
   gint n_elems;
-  GstVideoScalerHFunc hfunc;
-  GstVideoScalerVFunc vfunc;
+  GstVideoScalerHFunc hfunc = NULL;
+  GstVideoScalerVFunc vfunc = NULL;
   gint i;
 
   g_return_if_fail (src != NULL);
   g_return_if_fail (dest != NULL);
 
-  if (!get_functions (hscale, vscale, format, &hfunc, &vfunc, &n_elems, &width))
+  if (!get_functions (hscale, vscale, format, &hfunc, &vfunc, &n_elems, &width) || !vfunc || !hfunc)
     goto no_func;
 
 #define LINE(s,ss,i)  ((guint8 *)(s) + ((i) * (ss)))
