--- inkscape-0.48.4/src/widgets/desktop-widget.h~	2011-07-08 19:25:09.000000000 +0100
+++ inkscape-0.48.4/src/widgets/desktop-widget.h	2013-03-24 13:21:13.000000000 +0000
@@ -239,7 +239,7 @@ struct SPDesktopWidget {
 private:
     GtkWidget *tool_toolbox;
     GtkWidget *aux_toolbox;
-    GtkWidget *commands_toolbox,;
+    GtkWidget *commands_toolbox;
     GtkWidget *snap_toolbox;
 
     static void init(SPDesktopWidget *widget);
--- inkscape-0.48.4/src/libnrtype/FontFactory.h~	2011-07-08 19:25:09.000000000 +0100
+++ inkscape-0.48.4/src/libnrtype/FontFactory.h	2014-03-09 21:49:15.000000000 +0000
@@ -31,7 +31,8 @@
 #include <pango/pangowin32.h>
 #else
 #include <pango/pangoft2.h>
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #endif
 
 namespace Glib
