--- jpeg-9a/jmorecfg.h~	2013-09-17 08:20:20.000000000 +0100
+++ jpeg-9a/jmorecfg.h	2015-04-22 22:29:14.370819899 +0100
@@ -304,18 +304,13 @@ typedef void noreturn_t;
  */
 
 #ifndef HAVE_BOOLEAN
-#if defined FALSE || defined TRUE || defined QGLOBAL_H
-/* Qt3 defines FALSE and TRUE as "const" variables in qglobal.h */
-typedef int boolean;
+typedef unsigned char boolean;
 #ifndef FALSE			/* in case these macros already exist */
 #define FALSE	0		/* values of boolean */
 #endif
 #ifndef TRUE
 #define TRUE	1
 #endif
-#else
-typedef enum { FALSE = 0, TRUE = 1 } boolean;
-#endif
 #endif
 
 
