--- kdelibs-3.5.10/kio/kio/kdirwatch.cpp~	2006-07-22 09:16:37.000000000 +0100
+++ kdelibs-3.5.10/kio/kio/kdirwatch.cpp	2008-12-30 12:00:14.000000000 +0000
@@ -67,39 +67,7 @@
 #ifdef HAVE_INOTIFY
 #include <unistd.h>
 #include <fcntl.h>
-#include <sys/syscall.h>
-#include <linux/types.h>
-// Linux kernel headers are documented to not compile
-#define _S390_BITOPS_H
-#include <linux/inotify.h>
-
-static inline int inotify_init (void)
-{
-  return syscall (__NR_inotify_init);
-}
-
-static inline int inotify_add_watch (int fd, const char *name, __u32 mask)
-{
-  return syscall (__NR_inotify_add_watch, fd, name, mask);
-}
-
-static inline int inotify_rm_watch (int fd, __u32 wd)
-{
-  return syscall (__NR_inotify_rm_watch, fd, wd);
-}
-
-#ifndef  IN_ONLYDIR
-#define  IN_ONLYDIR 0x01000000 
-#endif
-
-#ifndef IN_DONT_FOLLOW
-#define IN_DONT_FOLLOW 0x02000000
-#endif
-
-#ifndef IN_MOVE_SELF
-#define IN_MOVE_SELF 0x00000800
-#endif
-
+#include <sys/inotify.h>
 #endif
 
 #include <sys/utsname.h>
--- kdelibs-3.5.10/kioslave/ftp/ftp.cc~	2008-08-19 19:18:12.000000000 +0100
+++ kdelibs-3.5.10/kioslave/ftp/ftp.cc	2009-10-12 18:59:47.000000000 +0100
@@ -876,7 +876,7 @@ int Ftp::ftpOpenPASVDataConnection()
   // The usual answer is '227 Entering Passive Mode. (160,39,200,55,6,245)'
   // but anonftpd gives '227 =160,39,200,55,6,245'
   int i[6];
-  char *start = strchr(ftpResponse(3), '(');
+  const char *start = strchr(ftpResponse(3), '(');
   if ( !start )
     start = strchr(ftpResponse(3), '=');
   if ( !start ||
@@ -931,7 +931,7 @@ int Ftp::ftpOpenEPSVDataConnection()
     return ERR_INTERNAL;
   }
 
-  char *start = strchr(ftpResponse(3), '|');
+  const char *start = strchr(ftpResponse(3), '|');
   if ( !start || sscanf(start, "|||%d|", &portnum) != 1)
     return ERR_INTERNAL;
 
--- kdelibs-3.5.10/config.h.in~	2008-08-20 17:05:20.000000000 +0100
+++ kdelibs-3.5.10/config.h.in	2009-10-12 18:50:05.000000000 +0100
@@ -1,5 +1,8 @@
 /* config.h.in.  Generated from configure.in by autoheader.  */
 
+/* Define if building universal (internal helper macro) */
+#undef AC_APPLE_UNIVERSAL_BUILD
+
 /* Define where aspell data dir is */
 #undef ASPELL_DATADIR
 
@@ -523,10 +526,10 @@
 /* Define to 1 if the system has the type `struct sockaddr_in6'. */
 #undef HAVE_STRUCT_SOCKADDR_IN6
 
-/* Define to 1 if `sin6_scope_id' is member of `struct sockaddr_in6'. */
+/* Define to 1 if `struct sockaddr_in6' is a member of `sin6_scope_id'. */
 #undef HAVE_STRUCT_SOCKADDR_IN6_SIN6_SCOPE_ID
 
-/* Define to 1 if `sa_len' is member of `struct sockaddr'. */
+/* Define to 1 if `struct sockaddr' is a member of `sa_len'. */
 #undef HAVE_STRUCT_SOCKADDR_SA_LEN
 
 /* Define if struct ucred is present from sys/socket.h */
@@ -769,6 +772,9 @@
 /* Define to the one symbol short name of this package. */
 #undef PACKAGE_TARNAME
 
+/* Define to the home page for this package. */
+#undef PACKAGE_URL
+
 /* Define to the version of this package. */
 #undef PACKAGE_VERSION
 
@@ -816,9 +822,17 @@
 /* Defined if compiling without arts */
 #undef WITHOUT_ARTS
 
-/* Define to 1 if your processor stores words with the most significant byte
-   first (like Motorola and SPARC, unlike Intel and VAX). */
-#undef WORDS_BIGENDIAN
+/* Define WORDS_BIGENDIAN to 1 if your processor stores words with the most
+   significant byte first (like Motorola and SPARC, unlike Intel). */
+#if defined AC_APPLE_UNIVERSAL_BUILD
+# if defined __BIG_ENDIAN__
+#  define WORDS_BIGENDIAN 1
+# endif
+#else
+# ifndef WORDS_BIGENDIAN
+#  undef WORDS_BIGENDIAN
+# endif
+#endif
 
 /* where rgb.txt is in */
 #undef X11_RGBFILE
