--- kdelibs-4.5.0/cmake/modules/FindDocBookXSL.cmake~	2010-07-06 07:54:45.000000000 +0100
+++ kdelibs-4.5.0/cmake/modules/FindDocBookXSL.cmake	2010-08-26 19:39:52.000000000 +0100
@@ -16,6 +16,7 @@ set (STYLESHEET_PATH_LIST
     share/sgml/docbook/xsl-stylesheets
     share/xml/docbook/stylesheet/nwalsh/current
     share/xml/docbook/stylesheet/nwalsh
+	share/xml/docbook-xsl
     share/xsl/docbook
 )
 
