--- libIDL-0.8.11/parser.y~	2007-01-01 23:41:34.000000000 +0000
+++ libIDL-0.8.11/parser.y	2008-11-06 14:52:08.000000000 +0000
@@ -97,6 +97,7 @@ static IDL_tree		IDL_binop_eval			(enum 
 							 IDL_tree b);
 static IDL_tree		IDL_unaryop_eval		(enum IDL_unaryop op,
 							 IDL_tree a);
+
 static IDL_tree		list_start			(IDL_tree a,
 							 gboolean filter_null);
 static IDL_tree		list_chain			(IDL_tree a,
@@ -607,7 +608,7 @@ struct_type:		z_props TOK_STRUCT
 			{ $<str>$ = "struct"; }
 			z_new_scope_catch '{'		{
 	g_hash_table_insert (__IDL_structunion_ht, $4, $4);
-	$$ = IDL_type_struct_new ($4, NULL);
+	$<tree>$ = IDL_type_struct_new ($4, NULL);
 }				member_list
 			'}' pop_scope			{
 	g_hash_table_remove (__IDL_structunion_ht, $4);
@@ -624,7 +625,7 @@ union_type:		z_props TOK_UNION
 				switch_type_spec
 			')' '{'				{
 	g_hash_table_insert (__IDL_structunion_ht, $4, $4);
-	$$ = IDL_type_union_new ($4, $7, NULL);
+	$<tree>$ = IDL_type_union_new ($4, $7, NULL);
 }				switch_body
 			'}' pop_scope			{
 	g_hash_table_remove (__IDL_structunion_ht, $4);
