--- librpcsecgss-0.19/include/rpcsecgss/rpc/rpc.h~	2009-07-29 17:24:50.000000000 +0100
+++ librpcsecgss-0.19/include/rpcsecgss/rpc/rpc.h	2011-10-29 20:07:00.000000000 +0100
@@ -71,6 +71,7 @@
 
 __BEGIN_DECLS
 
+#if 0
 /* Global variables, protected for multi-threaded applications.  */
 extern fd_set *__rpc_thread_svc_fdset (void) __attribute__ ((__const__));
 #define svc_fdset (*__rpc_thread_svc_fdset ())
@@ -93,6 +94,7 @@ extern struct pollfd **__rpc_thread_svc_
 
 extern int *__rpc_thread_svc_max_pollfd (void) __attribute__ ((__const__));
 #define svc_max_pollfd (*__rpc_thread_svc_max_pollfd ())
+#endif
 
 __END_DECLS
 
