--- ntp-4.2.6/ntpd/ntp_loopfilter.c~	2009-12-09 07:36:36.000000000 +0000
+++ ntp-4.2.6/ntpd/ntp_loopfilter.c	2009-12-14 16:46:33.000000000 +0000
@@ -517,7 +517,7 @@ local_clock(
 			ntv.modes = MOD_STATUS;
 		} else {
 #ifdef STA_NANO
-			ntv.modes = MOD_BITS | MOD_NANO;
+			ntv.modes = MOD_BITS | ADJ_NANO;
 #else /* STA_NANO */
 			ntv.modes = MOD_BITS;
 #endif /* STA_NANO */
--- ntp-4.2.8/ntpd/ntpd.c~	2014-12-19 11:58:06.000000000 +0000
+++ ntp-4.2.8/ntpd/ntpd.c	2014-12-29 21:30:55.485539928 +0000
@@ -105,8 +105,8 @@
 # include <grp.h>
 # include <pwd.h>
 #ifdef HAVE_LINUX_CAPABILITIES
-# include <sys/capability.h>
 # include <sys/prctl.h>
+# include <sys/capability.h>
 #endif /* HAVE_LINUX_CAPABILITIES */
 #if defined(HAVE_PRIV_H) && defined(HAVE_SOLARIS_PRIVS)
 # include <priv.h>
