--- ntp-4.2.4p7/ntpd/ntp_loopfilter.c~	2007-02-24 11:02:05.000000000 +0000
+++ ntp-4.2.4p7/ntpd/ntp_loopfilter.c	2009-11-10 17:26:04.000000000 +0000
@@ -568,7 +568,7 @@ local_clock(
 			time_t tstamp;
 
 #ifdef STA_NANO
-			ntv.modes = MOD_BITS | MOD_NANO;
+			ntv.modes = MOD_BITS | ADJ_NANO;
 #else /* STA_NANO */
 			ntv.modes = MOD_BITS;
 #endif /* STA_NANO */
@@ -893,7 +893,7 @@ loop_config(
 		pll_control = 1;
 		memset(&ntv, 0, sizeof(ntv));
 #ifdef STA_NANO
-		ntv.modes = MOD_BITS | MOD_NANO;
+		ntv.modes = MOD_BITS | ADJ_NANO;
 #else /* STA_NANO */
 		ntv.modes = MOD_BITS;
 #endif /* STA_NANO */
