--- pngcrush-1.7.25/Makefile~	2011-11-08 15:03:13.000000000 +0000
+++ pngcrush-1.7.25/Makefile	2012-03-05 19:55:52.000000000 +0000
@@ -21,7 +21,7 @@ RM = rm -f
 #CFLAGS = -I. -O2
 #CFLAGS = -I. -O3 -fomit-frame-pointer -Wall
 #CFLAGS = -I. -Os -fomit-frame-pointer -Wall
-CFLAGS = -I. -O3 -funroll-loops -fomit-frame-pointer -Wall -Wshadow
+CFLAGS = -I. -O3 -funroll-loops -fomit-frame-pointer -Wall -Wshadow -DPNGCRUSH_COUNT_COLORS
 #CFLAGS = -I${ZINC} -I. -O3 -fomit-frame-pointer -Wall
 # [note that -Wall is a gcc-specific compilation flag ("all warnings on")]
 
