--- qt-everywhere-opensource-src-4.7.0/src/3rdparty/webkit/JavaScriptCore/pcre/pcre.pri~	2010-09-10 10:05:22.000000000 +0100
+++ qt-everywhere-opensource-src-4.7.0/src/3rdparty/javascriptcore/JavaScriptCore/pcre/pcre.pri	2010-09-25 22:32:10.000000000 +0100
@@ -1,6 +1,9 @@
 # Perl Compatible Regular Expressions - Qt4 build info
 VPATH += $$PWD
-INCLUDEPATH += $$PWD $$OUTPUT_DIR/JavaScriptCore/tmp
+
+isEmpty(OUTPUT_DIR): OUTPUT_DIR = ../..
+
+INCLUDEPATH += $$PWD ../../JavaScriptCore/tmp
 DEPENDPATH += $$PWD
 
 SOURCES += \
--- qt-everywhere-opensource-src-4.7.0/src/3rdparty/webkit/JavaScriptCore/JavaScriptCore.pro~	2010-09-10 10:05:22.000000000 +0100
+++ qt-everywhere-opensource-src-4.7.0/src/3rdparty/webkit/Source/JavaScriptCore/JavaScriptCore.pro	2010-09-25 22:33:13.000000000 +0100
@@ -1,5 +1,8 @@
 # JavaScriptCore - qmake build info
 CONFIG += building-libs
+
+OUTPUT_DIR = ..
+
 include($$PWD/../WebKit.pri)
 include(JavaScriptCore.pri)
 
--- qt-everywhere-opensource-src-4.8.0/src/3rdparty/webkit/Source/WebKit.pri~	2012-02-10 16:56:48.000000000 +0000
+++ qt-everywhere-opensource-src-4.8.0/src/3rdparty/webkit/Source/WebKit.pri	2012-02-11 21:43:58.000000000 +0000
@@ -36,6 +36,8 @@ CONFIG(standalone_package): DEFINES *= N
 CONFIG += depend_includepath
 DEPENDPATH += $$OUT_PWD
 
+isEmpty(OUTPUT_DIR): OUTPUT_DIR = ../..
+
 DEFINES += BUILDING_QT__=1
 building-libs {
     win32-msvc*|win32-icc: INCLUDEPATH += $$PWD/JavaScriptCore/os-win32
--- qt-everywhere-opensource-src-4.8.0/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSGlobalObject.h~	2011-12-08 05:06:02.000000000 +0000
+++ qt-everywhere-opensource-src-4.8.0/src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSGlobalObject.h	2012-03-22 17:55:33.000000000 +0000
@@ -57,7 +57,7 @@ namespace JSC {
     
     class JSGlobalObject : public JSVariableObject {
     protected:
-        using JSVariableObject::JSVariableObjectData;
+        //using JSVariableObject::JSVariableObjectData;
 
         struct JSGlobalObjectData : public JSVariableObjectData {
             // We use an explicit destructor function pointer instead of a
--- qt-everywhere-opensource-src-4.8.6/configure~	2014-04-10 19:37:08.000000000 +0100
+++ qt-everywhere-opensource-src-4.8.6/configure	2015-07-24 21:55:37.696239453 +0100
@@ -7731,6 +7731,8 @@ case "$XPLATFORM" in
     *-g++*)
 	# Check gcc's version
 	case "$(${QMAKE_CONF_COMPILER} -dumpversion)" in
+	    5*)
+		;;
 	    4*|3.4*)
 		;;
             3.3*)
