--- taglib-1.5/taglib/taglib_export.h~	2008-02-04 15:14:46.000000000 +0000
+++ taglib-1.5/taglib/taglib_export.h	2008-04-16 21:56:59.000000000 +0100
@@ -26,7 +26,7 @@
 #ifndef TAGLIB_EXPORT_H
 #define TAGLIB_EXPORT_H
 
-#if defined(_WIN32) || defined(_WIN64)
+#if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__)
 #ifdef MAKE_TAGLIB_LIB
 #define TAGLIB_EXPORT __declspec(dllexport)
 #else
