--- webkitgtk-2.0.4/Source/ThirdParty/ANGLE/src/compiler/glslang.y~	2013-07-19 08:03:53.000000000 +0100
+++ webkitgtk-2.0.4/Source/ThirdParty/ANGLE/src/compiler/glslang.y	2013-08-15 18:13:05.000000000 +0100
@@ -41,12 +41,14 @@ WHICH GENERATES THE GLSL ES PARSER (glsl
 #define YYENABLE_NLS 0
 #define YYLTYPE_IS_TRIVIAL 1
 
-#define YYLEX_PARAM context->scanner
+//#define YYLEX_PARAM context->scanner
 %}
 
 %expect 1 /* One shift reduce conflict because of if | else */
-%pure-parser
+ //%pure-parser
+%define api.pure
 %parse-param {TParseContext* context}
+%lex-param {void* context}
 
 %union {
     struct {
--- webkitgtk-2.2.2/DerivedSources/ANGLE/glslang_tab.cpp~	2013-11-24 12:48:42.000000000 +0000
+++ webkitgtk-2.2.2/DerivedSources/ANGLE/glslang_tab.cpp	2013-11-24 12:49:16.000000000 +0000
@@ -1932,7 +1932,7 @@ yybackup:
   if (yychar == YYEMPTY)
     {
       YYDPRINTF ((stderr, "Reading a token: "));
-      yychar = yylex (&yylval, &yylloc, context, YYLEX_PARAM);
+      yychar = yylex (&yylval, &yylloc, context);
     }
 
   if (yychar <= YYEOF)
--- webkitgtk-2.2.2/Source/ThirdParty/ANGLE/src/compiler/glslang.y~	2013-11-24 11:19:38.000000000 +0000
+++ webkitgtk-2.2.2/Source/ThirdParty/ANGLE/src/compiler/glslang.y	2013-11-24 13:08:03.000000000 +0000
@@ -49,7 +49,7 @@ WHICH GENERATES THE GLSL ES PARSER (glsl
 %parse-param {TParseContext* context}
 %lex-param {void* context}
 %locations
-%lex-param {YYLEX_PARAM}
+ //lex-param {YYLEX_PARAM}
 
 %union {
 #define YYLTYPE TSourceLoc
